Documentation/development.md

```diff
@@ -0,0 +1,83 @@
+# CMO Development
+
+
+## Run CMO Locally
+### Assumptions
+* Should have a cluster running
+* Has setup Go development env
+
+### Build and Run CMO
+
+The `make` target `run-local` builds the `operator` and runs CMO locally
+
+```shell
+export KUBECONFIG=/path/to/kubeconfig/file
+make run-local
+```
+
+### Fix jsonnet
+
+**TL;DR:**
+1. build operator
+1. use `hack/local-cmo.sh`
+1. modify assets/yamls
+1. backport
+
+
+Sometimes we may not want to rebuild the operator. E.g. you may need
+to figure out how the stack behaves after applying a change to generated
+kubernetes manifest.  In other words, to modify the YAML in `assets/`
+directory directly without needing to change the `jsonnet/` or
+the `vendored` jsonnet files.
+
+In these cases it is easier to change those yaml files manually, run
+the operator (with those changed files) and later tweak `jsonnet`, than
+to figuring out which parts of jsonnet code you may need to change to
+fix an issue.
+
+It is easier to follow the following workflow
+Setup:
+1. Start openshift cluster ( e.g. using `cluster-bot` )
+1. `export KUBECONFIG=/path/to/kubeconfig`
+1. Build operator - `make operator`
+
+Repeat:
+1. Run `hack/local-cmo.sh` (**does not** build the operator)
+1. Amend yaml files
+1. ☝☝☝ _rinse and repeat_
+
+Finally:
+1. Port the changes back to `jsonnet` or vendored `jsonnet`
+
+
+
+
+## Updating vendored jsonnsets
+
+NOTE: `jb update <repo-url>/<jsonnet-subdir>` doesn't seem to work since it
```

Comment:
Is this a bug or is it expected? Is it related to https://github.com/jsonnet-bundler/jsonnet-bundler/issues/142?

Reply:
That could be the issue ... I haven't looked further into it. 

When I tried to update just `kube-prometheus` what I found was that it only updated the `kube-prometheus` and not any transitive dependencies. E.g. it didn't update the `kubernetes-mixin`. 

This was the workaround that worked.

Reply:
It seems that the same problem is being reported in that issue
```
What actually seems to be happening for transitive dependencies is that jb is retrieving the versions specified in the jsonnetfile.json in the dependency, rather than our local lock file.
```
I suggest we set up a ticket and fix this problem. We can still document the workaround temporarily and remove it once the problem with jsonnet-bundler is resolved.